platforms/core-configuration/groovy-support/src/main/java/org/gradle/api/internal/groovy/support/CompoundAssignmentExtensions.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.groovy.support;
+
+import org.jspecify.annotations.Nullable;
+
+/**
+ * This class provides catch-all support for {@code +=} overloading.
+ * <p>
+ * <b>This is a hidden public API</b>. Compiling Groovy code that depends on Gradle API may end up emitting references to methods of this class.
```

Comment:
❓Should we add it to the Public API definition then?
`build-logic-commons/basics/src/main/kotlin/gradlebuild/basics/PublicApi.kt`

Reply:
> Should we add it to the Public API definition then?

Unfortunately this is tied to exposing it in documentation, and we'll have to move it out of `internal` package.